tests/scripts/all.sh

```diff
@@ -4490,6 +4507,300 @@ component_test_aes_fewer_tables_and_rom_tables () {
     make test
 }
 
+component_test_block_cipher_no_decrypt_aesni_legacy () {
```

Comment:
It seems to me that newly added test components:
- `component_test_block_cipher_no_decrypt_aesni_legacy`
- `component_test_block_cipher_no_decrypt_aesni_use_psa`
- `component_test_block_cipher_no_decrypt_aesni_m32_legacy`
- `component_test_block_cipher_no_decrypt_aesni_m32_use_psa`

share most of the code. Only differences are unsetting some `PSA_WANT` symbols and/or different `CFLAGS`. Following what it has already been done with other components in the same `all.sh` file (ex: `build_full_minus_something_and_test_tls`) I suggest to:
- create a single helper functions which accepts 1 or 2 input parameters
    - For example `$1 should be `$USE_CRYPTO_CONFIG` and `$2` should be the `$SET_M32` flag, but you might come up with different ideas of course
- have all new test components to call this helper functions with proper parameters

Wdyt?